src/sdk/metrics/exporters/otlp.zig

```diff
@@ -85,39 +177,148 @@ fn attributesToProtobufKeyValueList(allocator: std.mem.Allocator, attributes: ?[
     }
 }
 
-fn sumDataPoints(allocator: std.mem.Allocator, comptime T: type, data_points: []DataPoint(T)) !std.ArrayList(pbmetrics.NumberDataPoint) {
-    var a = std.ArrayList(pbmetrics.NumberDataPoint).init(allocator);
+fn numberDataPoints(allocator: std.mem.Allocator, comptime T: type, data_points: []DataPoint(T)) !std.ArrayList(pbmetrics.NumberDataPoint) {
+    var a = try std.ArrayList(pbmetrics.NumberDataPoint).initCapacity(allocator, data_points.len);
     for (data_points) |dp| {
         const attrs = try attributesToProtobufKeyValueList(allocator, dp.attributes);
-        const number_dp = pbmetrics.NumberDataPoint{
+        a.appendAssumeCapacity(pbmetrics.NumberDataPoint{
```

Comment:
Small perf improvement: save allocations as we already know the number of items for the list.